### DISCUSSION TITLE
How to prevent a script with strategy=beforeInteractive from running during Cypress tests

### DISCUSSION THREAD

**Discussion by alvareztomas1** [vercel/next.js#73671] [Help]
### Summary

Hi people! I I have a this problem:

Basically, I need to load a script in the <head> tag of the root layout. This script should have the property strategy=beforeInteractive because it enables the use of an API in a newsletter subscription button.

The problem is that I also need it not to execute during Cypress tests. When it does, the tests break with the following error: [Error image](https://i.sstatic.net/9QdadlCK.png)

One way to achieve this is by checking if typeof window !== 'undefined' && !window.Cypress. However, this only works in components with "use client" (meaning I can't do it in the root layout). I tried moving the script to another .tsx file with the "use client" property, but the console throws a warning quoting the [Next.js documentation](https://nextjs.org/docs/messages/no-before-interactive-script-outside-document), which says that the strategy=beforeInteractive property should not be used in a client component.

Maybe strategy=beforeInteractive isn't necessary, and I could solve it by putting the script in another component with "use client" and checking that Cypress isn't running using the window object. As I mentioned, the script is necessary to allow users to interact with a newsletter subscription form to input an email.

### Additional information

```js
Root layout   
 <html lang="en">
      <head>
        <Script
          id="veroq-script"
          strategy="beforeInteractive"
          dangerouslySetInnerHTML={{
            __html: `
              var _veroq = _veroq || [];
              _veroq.push(['init', { api_key: '${process.env.NEXT_PUBLIC_VERO_API_KEY}' }]);
              (function () {
                var ve = document.createElement('script');
                ve.type = 'text/javascript';
                ve.async = true;
                ve.src = "//d3qxef4rp70elm.cloudfront.net/m.js";
                var s = document.getElementsByTagName('script')[0];
                s.parentNode.insertBefore(ve, s);
              })();
            `,
          }}
        />
      </head>
      <body className={`antialiased`}>
        <SessionProvider session={session}>
          <Navbar />
          {children}
          <Footer />
        </SessionProvider>
      </body>
    </html>




### Example

_No response_

**Comment by icyJoseph** [comment]
What if you don't pass this key, `NEXT_PUBLIC_VERO_API_KEY`, during CI/Cypress tests, and render the script conditionally? 

```jsx
{NEXT_PUBLIC_VERO_API_KEY && <YourScript />
```

Or variations of that? Perhaps with another env var?

**Reply by alvareztomas1** [reply]
I've solved it setting an env var during cypress tests with cross-env packagae and rendering the script conditionally. Thank you :)
